src/engine/render.tsx

```diff
@@ -194,11 +212,12 @@ const RenderEngine = ({
   config,
   sitemap,
   components,
+  LinkComponent,
 }: IsomerBaseSchema): React.ReactNode => {
   if (components && components.length > 0) {
     const collatedComponents = components.map(
       (component: IsomerComponent, idx: number) => {
-        return <div key={idx}>{getComponent(component)}</div>
+        return <div key={idx}>{getComponent(component, LinkComponent)}</div>
       },
```

Comment:
might be tripping but it seems like we have some circular logic between layouts and render engine
seemed to be a loop when I was tracing to understand who supplies the `LinkComponent` prop

Reply:
yea, there is circular or rather recursive logic - can update this in future PRs